rechunker/api.py

```diff
@@ -311,6 +312,43 @@ def rechunk(
     return Rechunked(executor, plan, source, intermediate, target)
 
 
+def parse_target_chunks_from_dim_chunks(ds, target_chunks):
+    """
+    Calculate ``target_chunks`` suitable for ``rechunker.rechunk()`` using chunks defined for
+    dataset dimensions (similar to xarray's ``.rechunk()``) .
+
+    - If a dimension is missing from ``target_chunks`` then use the full length from ``ds``.
+    - If a chunk in ``target_chunks`` is larger than the full length of the variable in ``ds``,
+      then, again, use the full length from the dataset.
+    - If a dimension chunk is specified as -1, again, use the full length from the dataset.
+
+    """
+    group_chunks = defaultdict(list)
+    for var in ds.variables:
+        for dim in ds[var].dims:
+            if dim in target_chunks.keys() and target_chunks[dim] <= len(ds[dim]):
+                group_chunks[var].append(target_chunks[dim])
+            elif (
+                dim in target_chunks.keys()
+                and (target_chunks[dim] > len(ds[dim]) or target_chunks[dim] < 0)
+            ) or not isinstance(ds[var].data, dask.array.Array):
```

Comment:
I hope I am not misreading this, but I think that If `target_chunks[dim]` is < 0 then the `if` condition on line 329 will evaluate as `True` and the `elif` condition will never be evaluated. Reversing the `if/elif` condition should fix this. Adding an explicit test for this wouldn't be a bad idea.  

That being said, I kind of dislike bundling the `isinstance()` check with the other conditions in the `elif`. I think that adding an explicit `elif` for the `isinstance()` check will make the code easier to read and easier to extend in the future if different types are supported. 

Finally, it might be worth considering whether it makes sense to put the inner loop in a separate function which would make it much easier to unit-test it. E.g.
 
``` python
def get_dim_chunk(ds, var, dim, target_chunks, existing_chunksizes):
    if not isinstance(ds[var].data, dask.array.Array):
        dim_chunk = len(ds[dim])
    elif dim in target_chunks.keys():
        if target_chunks[dim] > len(ds[dim]) or target_chunks[dim] < 0:
            dim_chunk = len(ds[dim])
        else:
            dim_chunk = target_chunks[dim]
    else:
        dim_chunk = existing_chunksizes[dim]
    return dim_chunk
```